far/vmenu.cpp

```diff
@@ -172,6 +171,47 @@ int find_nearest(std::ranges::contiguous_range auto const& Range, const int Pos,
 			: FindPos(drop(Pos), take(Pos) | reverse));
 }
 
+template <std::integral T>
+std::pair<T, T> Intersect(std::pair<T, T> A, std::pair<T, T> B)
+{
+	assert(A.first < A.second);
+	assert(B.first < B.second);
+
+	if (B.first < A.first)
+		std::swap(A, B);
+
+	if (A.second <= B.first)
+		return { {}, {} };
+
+	return { B.first, std::min(A.second, B.second) };
+}
+
+template <std::integral T>
+void MarkupSliceBoundaries(std::pair<T, T> Segment, std::ranges::input_range auto const& Slices, std::weakly_incrementable auto Markup)
+{
+	assert(Segment.first < Segment.second);
+
+	for (const auto Slice : Slices)
```

Comment:
`Slices` is a collection of `std::pair<int, int>`, so there is not much use in the extra indirection.

I tried to restrict the type of the input range element here, but all my attempts look quite ugly. They could be acceptable in a general-purpose library but not in this one-off context. If you can suggest something you would accept here, I'll be glad to use it and learn.

Reply:
I mentioned it mostly because static analyzers don't like by-value-for-loops at all and strongly suggest it's an error, without paying much attention to the underlying type.
So to see fewer bogus warnings it might be better to go with the flow.

Reply:
OK. Hopefully, compiler will optimize it out anyway.